### PR TITLE
+ Authorizations. Students cannot choose roles. Redirect to login for unauthorized users.

### PR DESCRIPTION
**Pull request by OlSarLang**

### CODE REVIEW

File: Controllers/SystemUserController.cs [hunk 1]
```diff
@@ -98,7 +98,6 @@ public async Task<IActionResult> Edit(string id, [Bind("Id,Name,StartDate,Descri
                 try
                 {
                     _context.Update(user);
-                    await _context.SaveChangesAsync();
```
Comment (Rocangus) [inline Controllers/SystemUserController.cs:101]: If we don't do this here, a DbUpdateConcurrencyException will break the entire page regardless and the try catch becomes redundant.

**Review by Rocangus — APPROVED** [review]
This looks okay enough but please check the comment

File: Controllers/SystemUserController.cs (hunk 1 above)
Comment (OlSarLang) [reply Controllers/SystemUserController.cs:101]: It breaks when it is in the Try method, but it works when it is outside.

Comment (Rocangus) [reply Controllers/SystemUserController.cs:101]: Okay, that's odd. I suppose this will have to do anyway,

**Pull request merged**